File: SupportDemo/app/src/main/java/cn/six/sup/rv/one_types_adapter/OneTypesAdapter.java
package cn.six.sup.rv.one_types_adapter;

import android.view.ViewGroup;

import cn.six.sup.rv.RvViewHolder;
import cn.six.sup.rv.OneAdapter;

public abstract class OneTypesAdapter<T> extends OneAdapter<T> {

    private TypesHelper<T> helper;

    public OneTypesAdapter(TypesHelper<T> helper) {
        super(-1);
        this.helper = helper;
    }

    @Override
    public int getItemViewType(int position) {
        return helper.getItemViewType(position, data.get(position));
    }

    @Override
    public RvViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        int layoutResId = helper.getLayoutResId(viewType);
        RvViewHolder holder = RvViewHolder.createViewHolder(parent, layoutResId);
        return holder;
    }
}

File: ConstraintLayoutDemo/app/src/main/java/ca/six/ctlay/utils/rv/compose/OneTypesAdapter.kt
package ca.six.ctlay.utils.rv.compose

import android.support.v7.widget.RecyclerView
import android.view.ViewGroup
import ca.six.advk.utils.rv.RvViewHolder

class OneTypesAdapter(private val data: List<ItemView>) : RecyclerView.Adapter<RvViewHolder>() {

    override fun getItemViewType(position: Int): Int {
        return data[position].viewType
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RvViewHolder {
        return RvViewHolder.createViewHolder(parent, viewType)
    }

    override fun onBindViewHolder(holder: RvViewHolder, position: Int) {
        data[position].bind(holder)
    }

    override fun getItemCount(): Int {
        return data.size
    }
}